modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -310,94 +276,58 @@ def get_files_info(ip, _rport, shares, info)
-            data: { shares: shares },
-            update: :unique_data
-          )
-
-          if datastore['SpiderShares']
-            begin
-              connect(versions: [1])
-              smb_login
-              get_files_info(ip, rport, shares, info)
-            rescue ::Rex::Proto::SMB::Exceptions::Error, Errno::ECONNRESET => e
-              print_error(
-                "Error when Spidering shares recursively (#{e}). This feature "\
-                'is only available with Rex client (SMB1 only) and the host '\
-                "probably doesn't support SMB1."
-              )
-            end
-          end
-
-          break if rport == 139
+        report_note(
+          host: ip,
+          proto: 'tcp',
+          port: rport,
+          type: 'smb.shares',
+          data: { shares: shares }, # We are now storing an array of hashes here, rather than an array of arrays. Will this cause issues further down the line?
+          update: :unique_data
+        )
+
+        if datastore['SpiderShares']
+          get_files_info(ip, shares)
         end
-      rescue ::Interrupt
-        raise $ERROR_INFO
-      rescue ::Rex::Proto::SMB::Exceptions::LoginError,
-             ::Rex::Proto::SMB::Exceptions::ErrorCode => e
-        print_error(e.message)
-        return if e.message =~ /STATUS_ACCESS_DENIED/
-      rescue Errno::ECONNRESET,
-             ::Rex::Proto::SMB::Exceptions::InvalidType,
-             ::Rex::Proto::SMB::Exceptions::ReadPacket,
-             ::Rex::Proto::SMB::Exceptions::InvalidCommand,
-             ::Rex::Proto::SMB::Exceptions::InvalidWordCount,
-             ::Rex::Proto::SMB::Exceptions::NoReply => e
-        vprint_error(e.message)
-        next if !shares.empty? && (rport == 139) # no results, try again
-      rescue Errno::ENOPROTOOPT
-        print_status('Wait 5 seconds before retrying...')
-        select(nil, nil, nil, 5)
-        retry
-      rescue ::Exception => e
-        next if e.to_s =~ /execution expired/
-        next if !shares.empty? && (rport == 139)
-
-        vprint_error("Error: '#{ip}' '#{e.class}' '#{e}'")
-      ensure
-        disconnect
       end
-
-      # if we already got results, not need to try on another port
-      return unless shares.empty?
+    rescue ::Interrupt
+      raise $ERROR_INFO
+    rescue Errno::ECONNRESET => e
+      vprint_error(e.message)
+    rescue Errno::ENOPROTOOPT
+      print_status('Wait 5 seconds before retrying...')
+      select(nil, nil, nil, 5)
+      retry
+    rescue Rex::ConnectionTimeout => e
+      print_error e.to_s
+      return
+    rescue StandardError => e
+      vprint_error("Error: '#{ip}' '#{e.class}' '#{e}'")
+    ensure
+      # Calling simple.client.disconnect! might not be needed here as we also call disconnect.
+      if simple && simple.client
+        simple.client.disconnect!
+      end
+      disconnect
     end
+
+    # if we already got results, not need to try on another port
+    return unless shares.empty?
```

Comment:
This code makes less sense without the 139/445 loop mechanism, but we're bringing that back anyway at least 😄